### ISSUE TITLE
Problem with marshaling and unmarshaling clusterName

### ISSUE THREAD

**Issue by LucaLanziani** [kubernetes/kubernetes#85394]
**What happened**:

Running automated CI on our cluster we generate the kubeadmcfg.yaml and we use an hash as clusterName, in one of our run we had:

```
clusterName: "433e40"
``` 

that running `kubeadm init phase kubeconfig scheduler --config=kubeadmcfg.yaml` results on a scheduler.conf containing:

```
apiVersion: v1
clusters:
- cluster:
....
  name: 433e40
```

The scheduler with this configuration will fail to start with:

```
2019-11-12T13:30:51.063658890+00:00 stderr F Error loading config file "/etc/kubernetes/scheduler.conf": v1.Config.Clusters: []v1.NamedCluster: v1.NamedCluster.Name: ReadString: expects " or n, but found 4, error found in #10 byte of ...|},"name":4.33e+42}],|..., bigger context ...|st:443"},"name":4.33e+42}],"contexts":[{"context":{"cluster":4.33e+|...
```

**What you expected to happen**:

Kubeadm to quote the cluster name or the reader to know that that is a string.

**How to reproduce it (as minimally and precisely as possible)**:

Set `clusterName` to "43e2" and run `kubeadm init phase kubeconfig scheduler --config=kubeadmcfg.yaml`

**Anything else we need to know?**:

This issue has already been discussed here: 

https://kubernetes.slack.com/archives/C13J86Z63/p1573566567090200

**Environment**:
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4", GitCommit:"67d2fcf276fcd9cf743ad4be9a9ef5828adc082f", GitTreeState:"clean", BuildDate:"2019-09-18T14:51:13Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by liggitt** [comment]
sounds similar to https://github.com/kubernetes/kubernetes/issues/82296, though I would expect the yaml generation step to have quoted the value (since quoting is required to distinguish it as a string)

**Comment by liggitt** [comment]
https://github.com/liggitt/kubernetes/commit/7bac7c115a98b23d4c7e8abccbbf0cc9280cb416 is a test case for the configuration writing method, and it demonstrates that it quotes numeric formatted names. Not sure what other transformations kubeadm is doing on the kubeconfig files.

/remove-sig api-machinery
/sig cluster-lifecycle
/area kubeadm

**Comment by neolit123** [comment]
/assign @rosti 
Rosti, could you please take a look ^.
thanks.

**Comment by liggitt** [comment]
actually, it looks like this *was* an issue with the yaml serializer, which happened to be resolved in 1.14.8+, 1.15.5+, and 1.16.2+

**Comment by liggitt** [comment]
added a test in #85401 to ensure this doesn't regress, but this is resolved in the releases listed above

/close
